map/src/login/HttpApiLogout.js

```diff
@@ -0,0 +1,233 @@
+
+    The concept:
+       
+        universal wrapper for axios/fetch functions
+            &&
+        global component for react-route nagivate from everywhere
+            ==
+        catch "auth-failed" from API and do react-route to "login-logout"
+
+        Notes:
+
+            login-logout is a login page with #logout hash to handle logout (loginLogoutForm)
+            default behavior: assume any 302-redirect as auth-failed => remote-logout
+            you can make any logic to detect API remote-logout answers
+
+    Usage as component for globalNavigate:
+
+        import { HttpApiLogout, globalNavigate } from '../login/HttpApiLogout';
+
+        <HttpApiLogout /> // add component inside BrowserRoute
+
+        call globalNavigate() everywhere you need
+
+    Usage as http-wrappers:
+
+        import { apiGet, apiPost } from '../login/HttpApiLogout';
+
+        call apiGet(url, options) instead of fetch() or axios() or axios.get()
+        call apiPost(url, data, options) instead of axios.post()
+
+        call quickNaNfix(badString) fix NaN(s) before JSON.parse() (Utils.js)
+
+    Read more details below.
+*/
+
+const loginLogoutForm = '/map/loginForm#logout';
+
+// global useNavigate()
+export let globalNavigate;
+
+// component to include inside <BrowserRoute>
+export const HttpApiLogout = () => {
+  globalNavigate = useNavigate();
+  return null;
+};
+
+/*
+    Universal wrapper replaces fetch() and axios() / axios.get() / axios.post()
+
+    Any 302 redirects cause navigate to login-logout (default)
+
+    Compatibility:
+
+        fetch: fully compatible, return original json/text/blob
+
+        axios: apiGet() support POST as { method: 'post' }
+        axios: apiGet() auto-detect-parse JSON into response.data
+        axios: apiGet() support single-parameter call like axios({ url, ... })
+        axios: apiGet() support { responseType: 'blob' } to force binary data return
+        
+        axios: apiPost() auto-detect JSON from post.data and stringify it
+        axios: apiPost() auto-detect post.data Content-Type (text/json/FormData)
+        axios: apiPost() support { params } to make query string for POST request
+        axios: apiPost() Content-Type header in options might be overriden by data auto-detect
+
+        NaN: fast method to fix NaN before JSON.parse
+            * default "ele":NaN is converted to "ele":NAN_MARKER
+            * others NaN are converted to null (that's OK because JSON never support NaN)
+            * available only for axios-like response.data: json()/text() will fail, use quickNaNfix() !
+
+    Options:
+
+        default { redirect: 'manual' } block redirect + do logout
+        use { redirect: 'error' } to block redirects + block logout
+        use { redirect: 'follow' } to allow redirects + block logout
+
+        any other options {} are passed to fetch() - { method: 'POST' } as example
+*/
+
+export async function apiGet(url, options = null) {
```

Comment:
apiGet, apiPost move to a separate util file ( mb module)

Reply:
HttpApiLogout was splitted by 2 parts